Sources/core/objects/Projectiles.cpp

```diff
@@ -0,0 +1,233 @@
+//TODO maybe it can apply froce depending on bullet speed but that should be considered        ApplyForce(DirToVDir[m->dMove]);
+void Projectile::MakeMovementPattern(VDir target,id_ptr_on<Human> someone)
+{
+    set_target_=1;
+    int x = target.x;
+    int y = target.y;
+    int abs_x = std::abs(x);
+    int abs_y = std::abs(y);
+    shooter_ = someone;
+    Dir facing = someone->GetDir();
+    if (abs_x == abs_y)
+    {
+        if (y<0)
+        {
+            if (facing == 0|| facing == 1)
+            {
+                if (x>0)
+                {
+                    MakeMovementLoops(abs_y,abs_x,D_RIGHT,D_UP);
+                }
+                else
+                {
+                    MakeMovementLoops(abs_y,abs_x,D_LEFT,D_UP);
+                }
+            }
+            else
+            {
+                if (x>0)
+                {
+                    MakeMovementLoops(abs_y,abs_x,D_UP,D_RIGHT);
+                }
+                else
+                {
+                    MakeMovementLoops(abs_y,abs_x,D_UP,D_LEFT);
+                }
+            }
+        }
+        else
+        {
+            if (facing == 0|| facing == 1)
+            {
+                if ( x > 0)
+                {
+                    MakeMovementLoops(abs_y,abs_x,D_RIGHT,D_DOWN);
+                }
+                else
+                {
+                    MakeMovementLoops(abs_y,abs_x,D_LEFT,D_DOWN);
+                }
+            }
+            else
+            {
+                if (x>0)
+                {
+                    MakeMovementLoops(abs_y,abs_x,D_DOWN,D_RIGHT);
+                }
+                else
+                {
+                    MakeMovementLoops(abs_y,abs_x,D_DOWN,D_LEFT);
+                }
+            }
+        }
+    }
+    else if (x >= 0 && y >= 0)
+    {
+       if (y > x)
+       {
+            MakeMovementLoops(abs_y,abs_x,D_DOWN,D_RIGHT);
+       }
+       else if (x > y)
+       {
+            MakeMovementLoops(abs_x,abs_y,D_RIGHT,D_DOWN);
+       }
+    }
+    else if(x <= 0 && y <= 0)
+    {   
+       if ( abs_y > abs_x  )
+       {
+           MakeMovementLoops(abs_y,abs_x,D_UP,D_LEFT);
+       }
+       else if ( abs_x > abs_y)
+       {
+           MakeMovementLoops(abs_x,abs_y,D_LEFT,D_UP);
+       }
+    }
+    else if (x < 0 && y > 0)
+    {
+        if (y > abs_x)
+        {
+             MakeMovementLoops(abs_y,abs_x,D_DOWN,D_LEFT);
+        }
+        else if (abs_x > abs_y)
+        {
+            MakeMovementLoops(abs_x,abs_y,D_LEFT,D_DOWN);
+        }
+    }
+    else if (x > 0 && y < 0)
+    {
+        if (abs_y > x)
+        {
+            MakeMovementLoops(abs_y,abs_x,D_UP,D_RIGHT);
+        }
+        else if (x > abs_y )
+        {
+            MakeMovementLoops(abs_x,abs_y,D_RIGHT,D_UP);
+        }
+        
+    }
+    else
+    {
+        qDebug() << "Critical error, MakeMovementPattern has reached unreachable place";
+        abort();
+    }
+    shooter_->Rotate(movement_[1]);
```

Comment:
It is another reason to not rotate the shooter in `MakeMovementPattern`.
Because new direction of the shooter is not directly connected to the movement pattern of the projectile.
![click_calculation](https://cloud.githubusercontent.com/assets/3407912/16546667/94bbf964-415a-11e6-9a6d-8e8aba06f484.png)
It should be done in `Shoot` function. Just check in that sector click was made and rotate shooter to according direction. Hint: if x > y then right or left, if y > x then it is up or down.